src/unistd.rs

```diff
@@ -1098,6 +1098,36 @@ pub fn close(fd: RawFd) -> Result<()> {
     Errno::result(res).map(drop)
 }
 
+#[cfg(target_os = "linux")]
+libc_bitflags! {
+    /// Options for close_range()
+    pub struct CloseRangeFlags : c_uint {
+        /// Unshare file descriptors before closing them.
+        CLOSE_RANGE_UNSHARE;
+        /// Set close-on-exec instead of closing file descriptors.
+        CLOSE_RANGE_CLOEXEC;
+    }
+}
+
+/// Close a range of raw file descriptors.
+/// [close_range(2)](https://man7.org/linux/man-pages/man2/close_range.2.html).
+#[cfg(target_os = "linux")]
+pub fn close_range(
+    first: RawFd,
+    last: RawFd,
+    flags: CloseRangeFlags,
+) -> Result<()> {
+    let res = unsafe {
+        libc::syscall(
```

Comment:
We shouldn't be using raw syscalls here.  They lack type safety, and they aren't portable.  Instead, you should add `close_range` in a PR to libc.  Then , once that merges, you can add support to Nix.  If you're ambitious, you might consider adding `closefrom` while you're at it.

Reply:
Ah, I see.  I was following the precedent of other functions in this file (eg: `gettid`, `execveat`, and `pivot_root`).

Reply:
For `gettid()` and `execveat()`, see this issue #2156, for `pivot_root`, there seems to be no libc wrapper for it, so we have to use the syscall directly.

-----
`close_range()` has a wrapper in glibc (now sure about other libcs), can you please add it to the upstream [libc](https://github.com/rust-lang/libc) crate and then we can add it in nix